bms-monitor.js

```diff
@@ -1,18 +1,90 @@
+const warningSettings ={
+  temperature: true,
+  soc: true,
+  chargeRate: true,
+};
+
+let language='en';
+
+const messages={
+  en:{
+    tempLow:'Warning: Approaching lower temperature limit',
+    tempHigh:'Warning: Approaching upper temperature limit',
+    socLow:'Warning: Approaching discharge',
+    socHigh: 'Warning: Approaching charge-peak',
+    chargeRateLow:'Warning: Approaching lower chargeRate limit',
+    chargerateHigh:'Warning: Approaching upper chargeRate limit',
+    tempOut:'Temperature is out of range!',
+    socOut:'State of Charge is out of range!',
+    chargeRateOut:'Charge rate is out of range!'
+
+  },
+
+  ge:
+  {
+    tempLow:'Warnung: Untere Temperaturgrenze nähert sich',
+    tempHigh:'Warnung: Die Temperaturobergrenze nähert sich',
+    socLow:'Warnung: Entladung steht kurz bevor',
+    socHigh:'Warnung: Ladespitze naht',
+    chargeRateLow:'Warnung: Untere Laderatengrenze nähert sich',
+    chargerateHigh:'Warnung: Die Obergrenze für die Laderate nähert sich',
+    tempOut:'Die Temperatur liegt außerhalb des zulässigen Bereichs!',
+    socOut:'Der Ladezustand liegt außerhalb des zulässigen Bereichs!',
+    chargeRateOut:'Laderate liegt außerhalb des zulässigen Bereichs!'
+  }
+}
 
-function batteryIsOk(temperature, soc, charge_rate) {
-  if (temperature < 0 || temperature > 45) {
-    console.log('Temperature is out of range!');
-    return false;
-  } else if (soc < 20 || soc > 80) {
-    console.log('State of Charge is out of range!');
-    return false;
-  } else if (charge_rate > 0.8) {
-    console.log('Charge rate is out of range!');
-    return false;
+function isOutOfRange(value, min, max, messagekey) {
+  if (value < min || value > max) {
+    console.log(messages[language][messagekey]);
+    return true;
+  }
+
+  return false;
+}
+function logWarning(value, limit, tolerance, messagekey) {
```

Comment:
Good abstraction :+1:  If this was repeated for every parameter, we would have to repeat the tests as well